data/transactions/logic/eval.go

```diff
@@ -628,15 +628,9 @@ func (ep *EvalParams) RecordAD(gi int, ad transactions.ApplyData) {
 		}
 		ep.available.createdAsas[aid] = struct{}{}
 	}
-	if aid := ad.ApplicationID; aid != 0 {
-		if ep.available == nil {
-			ep.available = ep.computeAvailability()
-		}
-		if ep.available.createdApps == nil {
-			ep.available.createdApps = make(map[basics.AppIndex]struct{})
-		}
-		ep.available.createdApps[aid] = struct{}{}
-	}
+	// we don't need to add ad.ApplicationID to createdApps, because that is
```

Comment:
I'm not sure this comment is even needed, but it's helpful for review purposes.

Reply:
I wasn't sure, but I think it's natural to wonder, because the clause above adds a newly created ASA to createdASAs, and those two maps have very similar purposes, but apps vs asa.